docs/dev-guide/using-sharrow.md

```diff
@@ -0,0 +1,73 @@
+
+# Using Sharrow
+
+This page will walk through an exercise of running a model with `sharrow`.
+
+
+## Getting the Code
+
+We'll assume that activitysim and sharrow have been installed in editable
+mode, per the [developer install instructions](install.md).
+
+The code to implement `sharrow` in `activitysim` is in the
+[sharrow-black](https://github.com/camsys/activitysim/tree/sharrow-black) branch
+of the [camsys/activitysim](https://github.com/camsys/activitysim) repository.
+If you have checked out from this repository directly, you can just switch to
+the correct branch:
+
+```sh
+cd activitysim
+git switch sharrow-black
+cd ..
+```
+
+If your local Git repository is configured with only the main ActivitySim
+parent repo as its remote, you can find the code in
+[PR #579](https://github.com/ActivitySim/activitysim/pull/579), accessible on the
+command line using the `gh` tool:
+
+```sh
+cd activitysim
+gh pr checkout 579  # use `gh auth login` first if needed
+cd ..
+```
+
+
+## Prototype MTC Examples
+
+Testing with sharrow requires two steps: test mode and production mode.
+
+In test mode, the code is run to compile all the spec files and
+ascertain whether the functions are working correctly.  Production mode
+can then just run the pre-compiled functions with sharrow, which is much
+faster.
+
+You can run both, plus the ActivitySim in "legacy" mode that does not use sharrow,
+as well as a reference implementation (version 1.0.4), all together using the
+"mini" dataset for testing in one workflow.
+
+```sh
+activitysim workflow sharrow-contrast/mtc_mini
+```
+
+Alternatively, you can use the full size skims.  To test this model with
+100k households and full skims (1475 zones), you can run the "mtc_full" workflow:
+
+```sh
+activitysim workflow sharrow-contrast/mtc_full
+```
+
+To use the full synthetic population as well, run the multiprocess workflow:
+
+```sh
+activitysim workflow sharrow-contrast/mtc_mp
+```
+
+Lastly, a comprehensive performance testing suite on the prototype MTC model
+(warning: this takes hours!)
+
+```sh
+activitysim workflow sharrow-contrast/mtc_comprehensive
+```
+
+All these performance tests assume you have sufficient RAM to run without chunking.
```

Comment:
On config changes and performance:  There have been minimal config changes, mostly the poorly performing utility spec files have been left intact if they work correctly.  Optimizing each spec is certainly worthwhile but outside the scope of our current task.  A small handful of spec changes have been made to ensure general compatibility, mostly moving commands from utility spec to preprocessors, usually because they execute another python function (e.g. reindex).  See new docs:
https://camsys.github.io/activitysim/sharrow-haircut-feature/dev-guide/using-sharrow.html#performance-considerations

On skim encoding:  it's never on if sharrow is not on.  If sharrow is enabled, digital encoding is controlled from network_los.yaml.  See new docs:
https://camsys.github.io/activitysim/sharrow-haircut-feature/dev-guide/using-sharrow.html#digital-encoding

On better user guide to settings in general (esp how to set up chunking):  yes we need this, but not part of this task/PR.  Sharrow is very focused on no-chunking model performance so far, we've done a bit of work looking at chunking compatibility but that is out of scope for this task.